rust/src/lib.rs

```diff
@@ -1606,64 +1607,21 @@ pub unsafe extern "C" fn zcashlc_seed_fingerprint(
     unwrap_exc_or(res, false)
 }
 
-/// Returns the most recent block height to which it is possible to reset the state
-/// of the data database.
+/// Rewinds the data database to at most the given height.
 ///
-/// # Safety
+/// If the requested height is greater than or equal to the height of the last scanned block, this
+/// function sets the `safe_rewind_ret` output parameter to `-1` and does nothing else.
 ///
-/// - `db_data` must be non-null and valid for reads for `db_data_len` bytes, and it must have an
-///   alignment of `1`. Its contents must be a string representing a valid system path in the
-///   operating system's preferred representation.
-/// - The memory referenced by `db_data` must not be mutated for the duration of the function call.
-/// - The total size `db_data_len` must be no larger than `isize::MAX`. See the safety
-///   documentation of pointer::offset.
-#[no_mangle]
-pub unsafe extern "C" fn zcashlc_get_nearest_rewind_height(
-    db_data: *const u8,
-    db_data_len: usize,
-    height: i32,
-    network_id: u32,
-) -> i32 {
-    #[allow(deprecated)]
-    let res = catch_panic(|| {
-        if height < 100 {
-            Ok(height)
-        } else {
-            let network = parse_network(network_id)?;
-            let db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
-            let height = BlockHeight::try_from(height)?;
-
-            match db_data.get_min_unspent_height() {
-                Ok(Some(best_height)) => {
-                    let first_unspent_note_height = u32::from(best_height);
-                    let rewind_height = u32::from(height);
-                    Ok(std::cmp::min(
-                        first_unspent_note_height as i32,
-                        rewind_height as i32,
-                    ))
-                }
-                Ok(None) => {
-                    let rewind_height = u32::from(height);
-                    Ok(rewind_height as i32)
-                }
-                Err(e) => Err(anyhow!(
-                    "Error while getting nearest rewind height for {}: {}",
-                    height,
-                    e
-                )),
-            }
-        }
-    });
-    unwrap_exc_or(res, -1)
-}
-
-/// Rewinds the data database to the given height.
+/// This procedure returns the height to which the database was actually rewound, or `-1` if no
+/// rewind was performed.
 ///
-/// If the requested height is greater than or equal to the height of the last scanned
-/// block, this function does nothing.
+/// If the requested rewind could not be performed, but a rewind to a different (greater) height
+/// would be valid, the `safe_rewind_ret` output parameter will be set to that value on completion;
+/// otherwise, it will be set to `-1`.
 ///
 /// # Safety
 ///
+/// - `safe_rewind_ret` must be non-null, aligned, and valid for writing a `uint32_t`.
```

Comment:
```suggestion
/// - `safe_rewind_ret` must be non-null, aligned, and valid for writing an `int64_t`.
```
Also move this to the bottom of the doc comment to match the parameter order.